build.gradle

```
@@ -94,50 +82,12 @@ repositories {
 sourceCompatibility = '1.6'
 targetCompatibility = '1.6'
 
-compileJava {
-    options.compilerArgs << "-s"
-    options.compilerArgs << "$projectDir/src/main/generated/java"
-
-    doFirst {
-        // make sure that directory exists
-        file(new File(projectDir, "/src/main/generated/java")).mkdirs()
-    }
-}
-
-compileTestJava {
-    options.compilerArgs << "-s"
-    options.compilerArgs << "$projectDir/src/test/generated/java"
-
-    doFirst {
-        // make sure that directory exists
-        file(new File(projectDir, "/src/test/generated/java")).mkdirs()
-    }
-}
-
-clean.doLast {
-    // clean-up directory when necessary
-    file(new File(projectDir, "/src/main/generated/java")).deleteDir()
-    file(new File(projectDir, "/src/test/generated/java")).deleteDir()
-}
-
-sourceSets {
-    main {
-        java {
-            srcDir "$projectDir/src/main/generated/java"
-        }
-    }
-
-    test {
-        java {
-            srcDir "$projectDir/src/test/generated/java"
-        }
-    }
-}
-
 dependencies {
     compileOnly 'com.squareup:javapoet:1.8.0'
 
     testCompileOnly 'com.squareup:javapoet:1.8.0'
-    testCompile group: 'junit', name: 'junit', version: '4.11'
+    testCompile group: 'junit', name: 'junit', version: '4.13.2'
     testCompile group: 'org.assertj', name: 'assertj-core', version: '3.6.2'
+    testAnnotationProcessor(getProject())
+    testAnnotationProcessor('com.squareup:javapoet:1.8.0')
```

I _think_ we can get rid of the parenthesis here, to make it consistent with the other declarations:

```suggestion
    testAnnotationProcessor getProject()
    testAnnotationProcessor 'com.squareup:javapoet:1.8.0'
```